CASMTRIAGE-8077: IUF workflow does not start after the re-installation of Keycloak (#5934)
Cray-HPE/docs-csm: operations/iuf/IUF.md

   check whether CFS is unable to start the configuration again for the node due to any other issue. Rectify the problem by referring to the
   [CFS troubleshooting guide](../../operations/configuration_management/Troubleshoot_CFS_Sessions_Failing_to_Start.md)

 1. IUF workflow does not start after the re-installation of Keycloak.

    - IUF waits for workflow to start. This step might not be completing because cray-nls stores the client-id and client-secret which is initialized when cray-nls pod is started if client_secret is modified after initializing such an error is possible.

>     - Look at the cray-nls pod logs. It should show that it is not able to validate "admin-client" id with keycloak as It was returning "unauthorized_client" new token request from cray-nls backend for "admin-client" id

Reviewer: Can you provide the exact error seen in the logs here? I think it would be helpful to provide exact output.
Also, you have a capitalized "I" that should be lowercase in: "keycloak as It was".

   check whether CFS is unable to start the configuration again for the node due to any other issue. Rectify the problem by referring to the
   [CFS troubleshooting guide](../../operations/configuration_management/Troubleshoot_CFS_Sessions_Failing_to_Start.md)

 1. IUF workflow does not start after the reinstallation of Keycloak.

>     - IUF stores the client-id and client-secret used to authorize with keycloak during initialization of IUF backend. When keycloak is reinstalled, the client-secret is regenerated and this client_secret stored by IUF is invalid.

Reviewer 2: this client_secret should be replaced with "the client_secret"

>     - IUF stores the client-id and client-secret used to authorize with keycloak during initialization of IUF backend. When keycloak is reinstalled, "the client_secret" is regenerated and this client_secret stored by IUF is invalid.

Reviewer: ```suggestion
    - IUF stores the client-id and client-secret used to authorize with keycloak during initialization of IUF backend. When keycloak is reinstalled, the `client_secret` is regenerated and the `client_secret` stored by IUF is no longer valid.
```

>     - The following error is seen in cray-nls pod(IUF backend) when IUF uses the stored client_id and client_secret to authorize with keycloak:

Reviewer: ```suggestion
    - The following error is seen in cray-nls pod (IUF backend) when IUF uses the stored `client_id` and `client_secret` to authorize with Keycloak:
```

>         ```bash

Reviewer 2: change ```bash to  ```text

         2025-04-14T06:25:58.143Z        ERROR   iuf/sessions_workflow_gen.go:210        Could not generate authToken Could not retrieve OIDC token:
         Expected 200 response but instead got 401
         {"error":"unauthorized_client","error_description":"Invalid client or Invalid client credentials"}
         ```

    - To start IUF workflows again, restart the cray-nls pods in argo namespace using the following command:

         ```bash
         kubectl rollout restart deployment cray-nls -n argo
         ```

>     - After cray-nls pods are restarted, reissue iuf command

Reviewer: ```suggestion
    - After cray-nls pods are restarted, reissue the iuf command that was previously being attempted.
```
